rfc/0068-kv-error-map-v2.md

```diff
@@ -0,0 +1,280 @@
+
+## Handling Error Attributes
+
+Error attributes beyond the retry attributes (discussed further down), which is utilized to indicate that a client must use the retry semantics of the particular error are not in scope for this document. A future RFC will provide details on additional attributes that may be added, and the behaviors they will provide. If in doubt if an attribute should be used, the SDK should use a conservative approach and favor backwards compatibility.
+
+# Error Attributes
+
+Error Categories act as the driving force behind the utility of the error map. Categories define the basic characteristics of an error condition so that even if the SDK does not understand a specific error code, it may determine the next course of action based on the error’s defined attributes.
+
+Attributes are the most important part of the error map as they convey how an error code should be handled. An error code may have more than a single attribute.
+
+- `item-only`: This attribute means that the error is related to a constraint failure regarding the item itself, i.e. the item does not exist, already exists, or its current value makes the current operation impossible. Retrying the operation when the item's value or status has changed may succeed.
+- `invalid-input`: This attribute means that a user's input was invalid because it violates the semantics of the operation, or exceeds some predefined limit.
+- `fetch-config`: The client's cluster map may be outdated and requires updating. The client should obtain a newer configuration.
+- `conn-state-invalidated`: The current connection is no longer valid. The client must reconnect to the server. Note that the presence of other attributes may indicate an alternate remedy to fixing the connection without a disconnect, but without special remedial action a disconnect is needed.
+- `auth`: The operation failed because the client failed to authenticate or is not authorized to perform this operation. Note that this error in itself does not mean the connection is invalid, unless conn-state-invalidated is also present.
+- `special-handling`: This error code must be handled specially. If it is not handled, the connection must be dropped.
+- `support`: The operation is not supported, possibly because the of server version, bucket type, or current user.
+- `temp`: This error is transient. Note that this does not mean the error is retriable.
+- `internal`: This is an internal error in the server.
+- `retry-now`: The operation may be retried immediately.
+- `retry-later`: The operation may be retried after some time.
+- `subdoc`: The error is related to the subdocument subsystem.
+- `dcp`: The error is related to the DCP subsystem.
+- `rate-limit`: The error is related to rate limiting being applied to the current connection or user.
```

Comment:
It's a shame the opportunity to add 'ambiguously committed mutation' as a concept here hasn't been seized by the server team.  This would have been useful in the past to allow SDK2 clients to handle DurabilityAmbiguous errors when bucket-level durability was raised.